server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -437,75 +441,107 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point, fina
         double lat = Double.NaN;
         double lon = Double.NaN;
         String geohash = null;
-        NumberFormatException numberFormatException = null;
+        String geojsonType = null;
+        ArrayList<Double> coordinates = null;
+        class NumberFormatExceptionHandler {
```

Comment:
This class exists only to keep the old behaviour that NumberFormatException is only handled at the end of parsing. However, there is no obvious reason why we would need that. It seems to me perfectly sufficient to throw the exception at the point that NumberFormatException arises, right? We have a number of other exceptions being thrown immediately during parsing, so why should this particular exception be treated differently. If we allow it to be thrown during parsing, this inner class can get replaced by a simple utility method.

Reply:
OK, after discussions with @iverase we decided to remove this. The work done in https://github.com/elastic/elasticsearch/pull/40447 should cover the requirements to finish parsing objects, so we don't need to capture exceptions anymore.